src/ClientServices/LaunchedProcess.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ClientServices/LaunchedProcess.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_client_services {
+
+using orbit_grpc_protos::ProcessInfo;
+using orbit_grpc_protos::ProcessToLaunch;
+
+ErrorMessageOr<LaunchedProcess> LaunchedProcess::LaunchProcess(
+    const orbit_grpc_protos::ProcessToLaunch& process_to_launch, ProcessClient* client) {
+  OUTCOME_TRY(ProcessInfo process_info, client->LaunchProcess(process_to_launch));
+  LaunchedProcess::State initial_state = process_to_launch.spin_at_entry_point()
+                                             ? LaunchedProcess::State::kSpinningAtEntryPoint
+                                             : LaunchedProcess::State::kExecutingOrExited;
+  return LaunchedProcess(initial_state, process_info);
+}
+
+LaunchedProcess::LaunchedProcess(State initial_state, orbit_grpc_protos::ProcessInfo process_info)
+    : state_(initial_state), process_info_(std::move(process_info)) {}
+
+ErrorMessageOr<void> LaunchedProcess::SuspendProcessSpinningAtEntryPoint(ProcessClient* client) {
+  ORBIT_CHECK(state_ == State::kSpinningAtEntryPoint);
+  OUTCOME_TRY(client->SuspendProcessSpinningAtEntryPoint(process_info_.pid()));
```

Comment:
I find passing `ProcessClient` to this class strange. I think this class should only used be used to keep state and the RPC should be done by the caller, instead of passing down the `ProcessClient`. In other words, this class shouldn't know about `ProcessClient`.